### PR TITLE
Exclude pb2.py files from format

### PR DESCRIPTION
**Pull request by dabacon**
Monkey banging at shell script.

### CODE REVIEW

File: check/format-incremental [hunk 1]
```diff
@@ -66,6 +66,17 @@ if [ -z "${rev}" ]; then
 fi
 echo "Comparing against revision '${rev}'." >&2
 
+# Copy over .yapfignore or use local version. Cleanup if copying.
+ignore_exists='[ -f .yapfignore ]'
+if ! $ignore_exists; then
+  function cleanup {
+    rm .yapfignore
```
Comment (Strilanc) [inline check/format-incremental]: Can we pass this file in as a command line parameter, instead of by copying and deleting?

File: .yapfignore [hunk 2]
```diff
@@ -0,0 +1 @@
+**/*_pb2.py
```
Comment (vtomole) [inline .yapfignore]: newline needed

File: dev_tools/conf/.yapfignore [hunk 3]
```diff
@@ -0,0 +1 @@
+**/*_pb2.py
```
Comment (vtomole) [inline dev_tools/conf/.yapfignore]: newline

File: check/format-incremental [hunk 4]
```diff
@@ -66,6 +66,17 @@ if [ -z "${rev}" ]; then
...
+  }
+  trap cleanup EXIT;
+fi
+cp -n dev_tools/conf/.yapfignore .yapfignore
```
Comment (vtomole) [inline check/format-incremental]: Oh this is where the 2 files are comming from.

File: check/format-incremental (hunk 1 above)
Comment (dabacon) [reply check/format-incremental]: yapf doesn't allow a config path to be passed in :(

**Comment by dabacon** [comment]
PTAL.  Just filtered it out of changed files directly.

File: check/format-incremental [hunk 5]
```diff
@@ -68,7 +68,7 @@ echo "Comparing against revision '${rev}'." >&2
 
 # Get the _test version of changed python files.
 needed_changes=0
-changed_files=$(git diff --name-only ${rev} -- | grep "\.py$")
+changed_files=$(git diff --name-only ${rev} -- | grep "\.py$" | grep -v "pb2\.py$")
```
Comment (Strilanc) [inline check/format-incremental]: You should also include the underscore. I can imagine a file accidentally ending in `pb2`, but `_pb2` not so much.

Comment (dabacon) [reply check/format-incremental]: done

**Pull request merged**